src/core/update_initial_conditions.jl

```diff
@@ -0,0 +1,326 @@
+) where {T <: PSY.ThermalGen}
+    _make_initial_conditions!(
+        container,
+        devices,
+        ICKey(DeviceStatus, T),
+        _make_initial_condition_active_power,
+        _get_active_power_status_value,
+    )
+
+    return
+end
+
+function output_init(
+    container::InitialConditions,
+    devices::IS.FlattenIteratorWrapper{T},
+) where {T <: PSY.ThermalGen}
+    _make_initial_conditions!(
+        container,
+        devices,
+        ICKey(DevicePower, T),
+        _make_initial_condition_active_power,
+        _get_active_power_output_value,
+    )
+
+    return
+end
+
+function duration_init(
+    container::InitialConditions,
+    devices::IS.FlattenIteratorWrapper{T},
+) where {T <: PSY.ThermalGen}
+    for key in (ICKey(TimeDurationON, T), ICKey(TimeDurationOFF, T))
+        _make_initial_conditions!(
+            container,
+            devices,
+            key,
+            _make_initial_condition_active_power,
+            _get_active_power_duration_value,
+            TimeStatusChange,
+        )
+    end
+
+    return
+end
+
+######################### Initialize Functions for Storage #################################
+# TODO: This IC needs a cache for Simulation over long periods of tim
+function storage_energy_init(
+    container::InitialConditions,
+    devices::IS.FlattenIteratorWrapper{T},
+) where {T <: PSY.Storage}
+    key = ICKey(EnergyLevel, T)
+    _make_initial_conditions!(
+        container,
+        devices,
+        key,
+        _make_initial_condition_energy,
+        _get_energy_value,
+    )
+
+    return
+end
+
+######################### Initialize Functions for Hydro #################################
+function status_init(
+    container::InitialConditions,
+    devices::IS.FlattenIteratorWrapper{T},
+) where {T <: PSY.HydroGen}
+    _make_initial_conditions!(
+        container,
+        devices,
+        ICKey(DeviceStatus, T),
+        _make_initial_condition_active_power,
+        _get_active_power_status_value,
+        # Doesn't require Cache
+    )
+end
+
+function output_init(
+    container::InitialConditions,
+    devices::IS.FlattenIteratorWrapper{T},
+) where {T <: PSY.HydroGen}
+    _make_initial_conditions!(
+        container,
+        devices,
+        ICKey(DevicePower, T),
+        _make_initial_condition_active_power,
+        _get_active_power_output_value,
+        # Doesn't require Cache
+    )
+
+    return
+end
+
+function duration_init(
+    container::InitialConditions,
+    devices::IS.FlattenIteratorWrapper{T},
+) where {T <: PSY.HydroGen}
+    for key in (ICKey(TimeDurationON, T), ICKey(TimeDurationOFF, T))
+        _make_initial_conditions!(
+            container,
+            devices,
+            key,
+            _make_initial_condition_active_power,
+            _get_active_power_duration_value,
+            TimeStatusChange,
+        )
+    end
+
+    return
+end
+
+# TODO: This IC needs a cache for Simulation over long periods of time
```

Comment:
This comment can be removed now. It was there before but if you can eliminate it would be great.